### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by JonathanHUnity**
Various improvements and features, including support for:

- partial types
- structs
- multiple xml docs on same member
- operators
- constructors
- generics
- extension methods
- optional parameters
- delegates
- 'set' on methods

### CODE REVIEW

File: Unity.DocTool.XMLDocHandler.Tests/TestTypes/ClassWithOperator.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+﻿namespace Unity.DocTool.XMLDocHandler.Tests.TestTypes
+{
+    class ClassWithOperator
+    {
+        /// <summary>
+        /// Plus Operator
+        /// </summary>
+        public static int operator +(ClassWithOperator classWithOperator, int other)
```
Comment (adrianoc-unity3d) [inline Unity.DocTool.XMLDocHandler.Tests/TestTypes/ClassWithOperator.cs]: I think we should add tests for *implicit*/*explicit* user conversion operators as well

File: Unity.DocTool.XMLDocHandler.Tests/TestTypes/ClassWithConstructor.cs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+﻿namespace Unity.DocTool.XMLDocHandler.Tests.TestTypes
+{
+    class ClassWithConstructor
```
Comment (adrianoc-unity3d) [inline Unity.DocTool.XMLDocHandler.Tests/TestTypes/ClassWithConstructor.cs]: Static ctors?

File: Unity.DocTool.XMLDocHandler/XMLDocHandler.cs [hunk 3]
```diff
@@ -209,41 +220,104 @@ private string SignatureFor(ISymbol member)
...
+            {
+                return TypeReferenceXml(namedTypeSymbol);
+            }
+
+            var sourceTypeParameterSymbol = typeSymbol as ITypeParameterSymbol;
+            if (sourceTypeParameterSymbol != null)
+            {
+                return TypeParameterXml(sourceTypeParameterSymbol, includeConstraints, true);
+            }
+            throw new NotImplementedException("Unsupported typeSymbol");
```
Comment (adrianoc-unity3d) [inline Unity.DocTool.XMLDocHandler/XMLDocHandler.cs]: just to make our lifes easier I guess we should include information about the *typeSymbol*  in the exception.

File: Unity.DocTool.XMLDocHandler.Tests/TestTypes/ClassWithConstructor.cs (hunk 2 above)
Comment (adrianoc) [reply Unity.DocTool.XMLDocHandler.Tests/TestTypes/ClassWithConstructor.cs]: I guess developer may still want to document what the static ctor is doing no?

Comment (JonathanHUnity) [reply Unity.DocTool.XMLDocHandler.Tests/TestTypes/ClassWithConstructor.cs]: Sure, but where would that documentation show up? I don't see any static constructors in MSDN for example, I think any relevant stuff goes on the class itself.

**Pull request merged**